src/lib/dialog/dialog.ts

```diff
@@ -40,6 +40,29 @@ import {MatDialogRef} from './dialog-ref';
 
 export const MAT_DIALOG_DATA = new InjectionToken<any>('MatDialogData');
 
+/** Injection token that can be used to specify default dialog options. */
+export const MAT_DIALOG_DEFAULT_OPTIONS =
+    new InjectionToken<MatDialogConfig>('mat-dialog-default-options');
+
+/** @docs-private */
+export const MAT_DIALOG_DEFAULT_OPTIONS_PROVIDER = {
```

Comment:
A couple of notes on this one:
1. You don't have to define a provider object for it, you can just put it inline into the `providers`. We didn't do the same for the scroll strategy, because it has a bit more boilerplate.
2. Why are you providing these values here? If you use a `new MatDialogConfig()` it'll give you an object with the defaults provided already.

Reply:
I tried providing `new MatDialogConfig()` before but when compiling Angular outputs this error: `Error during template compile of 'MatDialogModule' Function calls are not supported in decorators but 'MatDialogConfig' was called.` 

I also tried providing with `useClass: MatDialogConfig` but then providing a `MAT_DIALOG_DEFAULT_OPTIONS` in another module with `useValue` does not override the `MatDialogModule` defaults.

Reply:
Sorry for the delay. `useClass` definitely won't work, because Angular will try to instantiate it as if it a service. What about doing something like this:

```ts
const defaultConfig = new MatDialogConfig();

@NgModule({
  ...
  providers: [
    {provide: MAT_DIALOG_DEFAULT_OPTIONS, useValue: defaultConfig}
  ]
})
export class MatDialogModule {}
```

Reply:
That didn't work, still getting the same error. 

I was thinking maybe it would be better to provide `MAT_DIALOG_DEFAULT_OPTIONS` the way I am doing right now and just define `MatDialogConfig` without any preset fields. This way we can avoid duplicating the same values on both sides and also when a user does this :
```
let config = new MatDialogConfig(); 
config.hasBackdrop = false;
dialogs.open(Dialog, config);
```
The default `MatDialogConfig` fields will not override their own provided default options.

Reply:
In that case it's best to go with what you had initially. I'm a little iffy about having the triple spread that you had before, but something like this should be fine for applying the defaults:

```ts
config = _applyConfigDefaults(this._defaultOptions || new MatDialogConfig(), config);
```

Reply:
That looks way neater than what I had previously. Thank you for the help